x/sysproxy/sysproxy_darwin.go

```diff
@@ -0,0 +1,101 @@
+//go:build darwin
+
+package sysproxy
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+func SetProxy(ip string, port string) error {
+	// Execute macOS specific commands to set proxy
+	// Get the active network interface
+	activeInterface, err := getActiveNetworkInterfaceMacOS()
+	if err != nil {
+		return err
+	}
+
+	// Set the web proxy and secure web proxy
+	if err := setProxyMacOS("web", activeInterface, ip, port); err != nil {
+		return err
+	}
+	if err := setProxyMacOS("secureweb", activeInterface, ip, port); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func UnsetProxy() error {
+	// Execute macOS specific commands to unset proxy
+	// Get the active network interface
+	activeInterface, err := getActiveNetworkInterfaceMacOS()
+	if err != nil {
+		return err
+	}
+
+	// Set the web proxy and secure web proxy
+	if err := removeProxyMacOS("web", activeInterface); err != nil {
+		return err
+	}
+	if err := removeProxyMacOS("secureweb", activeInterface); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// getActiveNetworkInterface finds the active network interface using shell commands.
+func getActiveNetworkInterfaceMacOS() (string, error) {
+	cmd := "sh -c \"networksetup -listnetworkserviceorder | grep `route -n get 0.0.0.0 | grep 'interface' | cut -d ':' -f2` -B 1 | head -n 1 | cut -d ' ' -f2\""
```

Comment:
Just double check, do we need the `sh -c` here since we will append `"bash", "-c"` below?